src/pages/mentorship/index.tsx

```diff
@@ -1,13 +1,30 @@
 // path: /mentorship
+import { GetServerSideProps } from 'next';
 
-import { Typography } from '@mui/material';
+import { MentorshipPageProps } from '@utils/types';
+import { fetchData } from 'lib/api';
 
-const MentorshipPage = () => {
-  return (
-    <div>
-      <Typography variant="h4">Welcome to the Mentorship Page</Typography>
-    </div>
-  );
+const MentorshipPage = ({ mentorship }: MentorshipPageProps) => {
```

Comment:
PageProps should be in the page, it belongs here and won't be needed in other pages